src/acom_music_box/plot_output.py

```diff
@@ -0,0 +1,166 @@
+import logging
+import matplotlib.pyplot as plt
+import mplcursors
+import subprocess
+import os
+import tempfile
+
+logger = logging.getLogger(__name__)
+
+class PlotOutput:
+  """
+  A class to handle plotting operations for a DataFrame, including plotting species
+  concentrations over time using gnuplot or matplotlib.
+  
+  This class manages the plotting tool, species list, and data output formats based on
+  the provided arguments, ensuring valid paths and creating necessary directories.
+
+  Attributes
+  ----------
+  df : pandas.DataFrame
+      The DataFrame to be plotted.
+  args : argparse.Namespace
+      Command-line arguments or configurations specifying plot options.
+  species_list : list
+      A list of species to plot.
+      
+  Examples
+  --------
+  >>> import pandas as pd
+  >>> from argparse import Namespace
+  >>> df = pd.DataFrame({
+  ...     'time': [0, 1, 2],
+  ...     'CONC.A.mol m-3': [1, 2, 3],
+  ...     'CONC.B.mol m-3': [4, 5, 6],
+  ...     'CONC.C.mol m-3': [7, 8, 9]
+  ... })
+  >>> args = Namespace(plot='CONC.A,CONC.B', plot_tool='matplotlib')
+  >>> plot_output = PlotOutput(df, args)
+  >>> plot_output.plot()
+  """
+
+  def __init__(self, df, args):
+    """
+    Initialize the PlotOutput class with a DataFrame and configuration arguments.
+
+    Parameters
+    ----------
+    df : pandas.DataFrame
+        The DataFrame containing the data to be output.
+    args : argparse.Namespace
+        Arguments specifying the plot configuration, such as plot tool and species list.
+    """
+
+    self.df = df.copy(deep=True)
+    self.args = args
+    self.species_list = self._format_species_list(self.args.plot.split(',') if self.args.plot else None)
+  
+
+  def _format_species_list(self, species_list):
+    """
+    Format the species list for plotting.
+
+    This method formats the species list for plotting by adding the 'CONC.' prefix
+    to each species name if it is not already present.
+
+    Parameters
+    ----------
+    species_list : list
+        A list of species to plot.
+
+    Returns
+    -------
+    list
+        A formatted list of species for plotting.
+    """
+
+    plot_list = None
+    if species_list is not None:
+      plot_list = []
+      for species in species_list:
+          species = species.strip()
+          if 'CONC.' not in species:
+              species = f'CONC.{species}'
+          plot_list.append(species)
+      
+    return plot_list
+
+  def _plot_with_gnuplot(self):
+      """
+      Plot the specified species using gnuplot.
+      """
+      # Prepare columns and data for plotting
+      columns = ['time'] + self.species_list
+      data_to_plot = self.df[columns]
+
+      data_csv = data_to_plot.to_csv(index=False)
+
+      try:
+          with tempfile.NamedTemporaryFile(suffix='.csv', mode='w+', delete=True) as data_file:
```

Comment:
@adityakiran1423 this named temporary file gets deleted automatically, even if any exception is thrown. `delete=True` was added, but this is redundant because that's the default behavior

Reply:
An oversight on my side, thanks for pointing out

Reply:
An oversight on my side, thanks for highlighting!